### PR TITLE
Fixed Phase Rewind not Behaving Well

### PR DESCRIPTION
**Pull request by DBmaoha**
Phase Rewind will now have a longer distance and mostly won't stuck as It be like in vanilla
Map Hack won't show sonar pulse to other players anymore, also be like in vanilla

### CODE REVIEW

**Review by uniboi — APPROVED** [review]
https://user-images.githubusercontent.com/64006268/191799273-c78f5b28-1a8f-430c-8917-d221b31172c6.mp4

I think this mimics vanilla perfectly
https://user-images.githubusercontent.com/64006268/191799318-d15f52f2-942f-4800-b5a3-38f063a3c055.mp4

**Comment by Erlite** [comment]
Do you have a video to compare before your PR? Just curious what the difference is.

**Comment by uniboi** [comment]
look at the filenames lmao the first on is vanilla the second one is from the pr

**Comment by Erlite** [comment]
No I mean in Northstar before your PR, unless that's what Vanilla is?

**Comment by uniboi** [comment]
https://user-images.githubusercontent.com/64006268/191807056-96957802-aa6a-4285-9016-1a626bc3b478.mp4

there's a notable difference to the state phase rewind is currently in northstar

**Comment by Erlite** [comment]
Awesome work!

File: Northstar.CustomServers/mod/scripts/vscripts/burnmeter/_burnmeter.gnut [hunk 1]
```diff
@@ -470,18 +473,66 @@ void function PlayerUsesPhaseRewindBurncardThreaded( entity player )
...
+	if( PutEntityInSafeSpot( player, player, null, < baseOrigin.x-severity, baseOrigin.y, baseOrigin.z >, baseOrigin ) )
+        return
+
+	if( PutEntityInSafeSpot( player, player, null, < baseOrigin.x, baseOrigin.y, baseOrigin.z+severity >, baseOrigin ) )
+        return
+
+	if( PutEntityInSafeSpot( player, player, null, < baseOrigin.x, baseOrigin.y, baseOrigin.z-severity >, baseOrigin ) )
+        return
+
+    return PutPhaseRewindedPlayerInSafeSpot( player, severity+5 )
```
Comment (ASpoonPlaysGames) [inline Northstar.CustomServers/mod/scripts/vscripts/burnmeter/_burnmeter.gnut:535]:
```suggestion
	if ( PutEntityInSafeSpot( player, player, null, < baseOrigin.x, baseOrigin.y + severity, baseOrigin.z >, baseOrigin ) )
		return

	if ( PutEntityInSafeSpot( player, player, null, < baseOrigin.x, baseOrigin.y - severity, baseOrigin.z >, baseOrigin ) )
		return

	if ( PutEntityInSafeSpot( player, player, null, < baseOrigin.x + severity, baseOrigin.y, baseOrigin.z >, baseOrigin ) )
		return

	if ( PutEntityInSafeSpot( player, player, null, < baseOrigin.x - severity, baseOrigin.y, baseOrigin.z >, baseOrigin ) )
        	return

	if ( PutEntityInSafeSpot( player, player, null, < baseOrigin.x, baseOrigin.y, baseOrigin.z + severity >, baseOrigin ) )
		return

	if ( PutEntityInSafeSpot( player, player, null, < baseOrigin.x, baseOrigin.y, baseOrigin.z - severity >, baseOrigin ) )
		return

	return PutPhaseRewindedPlayerInSafeSpot( player, severity + 5 )

```
formatting things

File: Northstar.CustomServers/mod/scripts/vscripts/burnmeter/_burnmeter.gnut [hunk 2]
```diff
@@ -470,18 +473,66 @@ void function PlayerUsesPhaseRewindBurncardThreaded( entity player )
...
+		float moveTime = PHASE_REWIND_PATH_SNAPSHOT_INTERVAL + 0.1
+		//if( i == 0 )
+		//	moveTime = PHASE_REWIND_PATH_SNAPSHOT_INTERVAL
+		mover.NonPhysicsMoveTo( positions[ i ].origin, moveTime, 0, 0 )
+		mover.NonPhysicsRotateTo( positions[ i ].angles, moveTime, 0, 0 )
 		wait PHASE_REWIND_PATH_SNAPSHOT_INTERVAL
 	}
 	player.SetVelocity( <0, 0, 0> )
+	if( positions[0].wasCrouched )
+		player.SetOrigin( player.GetOrigin() + < 0,0,20 > )
```
Comment (ASpoonPlaysGames) [inline Northstar.CustomServers/mod/scripts/vscripts/burnmeter/_burnmeter.gnut:507]:
```suggestion
		player.SetOrigin( player.GetOrigin() + < 0, 0, 20 > )
```

File: Northstar.CustomServers/mod/scripts/vscripts/burnmeter/_burnmeter.gnut [hunk 3]
```diff
@@ -470,18 +473,66 @@ void function PlayerUsesPhaseRewindBurncardThreaded( entity player )
...
 
 	for ( int i = positions.len() - 1; i > -1; i-- )
 	{
-		mover.NonPhysicsMoveTo( positions[ i ].origin, PHASE_REWIND_PATH_SNAPSHOT_INTERVAL, 0, 0 )
-		mover.NonPhysicsRotateTo( positions[ i ].angles, PHASE_REWIND_PATH_SNAPSHOT_INTERVAL, 0, 0 )
+		//print( positions[ i ].origin )
+		// should looks better?
+		float moveTime = PHASE_REWIND_PATH_SNAPSHOT_INTERVAL + 0.1
+		//if( i == 0 )
+		//	moveTime = PHASE_REWIND_PATH_SNAPSHOT_INTERVAL
```
Comment (ASpoonPlaysGames) [inline Northstar.CustomServers/mod/scripts/vscripts/burnmeter/_burnmeter.gnut:500]:
```suggestion
```
Can probably remove these lines

File: Northstar.CustomServers/mod/scripts/vscripts/burnmeter/_burnmeter.gnut [hunk 4]
```diff
@@ -470,18 +473,66 @@ void function PlayerUsesPhaseRewindBurncardThreaded( entity player )
...
 	player.SetPredictionEnabled( false )
 	PhaseShift( player, 0.0, positions.len() * PHASE_REWIND_PATH_SNAPSHOT_INTERVAL * 1.5 )
+	EmitSoundOnEntityOnlyToPlayer( player, player, "pilot_phaserewind_1p" )
+	EmitSoundOnEntityExceptToPlayer( player, player, "pilot_phaserewind_3p" )
 
 	for ( int i = positions.len() - 1; i > -1; i-- )
 	{
-		mover.NonPhysicsMoveTo( positions[ i ].origin, PHASE_REWIND_PATH_SNAPSHOT_INTERVAL, 0, 0 )
-		mover.NonPhysicsRotateTo( positions[ i ].angles, PHASE_REWIND_PATH_SNAPSHOT_INTERVAL, 0, 0 )
+		//print( positions[ i ].origin )
```
Comment (ASpoonPlaysGames) [inline Northstar.CustomServers/mod/scripts/vscripts/burnmeter/_burnmeter.gnut:496]:
```suggestion
```
Debug print can probably be removed

**Review by ASpoonPlaysGames — CHANGES_REQUESTED** [review]
Formatting things + removing now-unneeded lines

**Comment by GeckoEidechse** [comment]
@uniboi do you wanna retest after new changes, in particular 136a75f9e40668249dd38f6c21cb505259569a27 ?

Code formatting looks fine from what I can tell, so assuming @ASpoonPlaysGames did code (logic) review we could merge it after testing. ^^

**Comment by uniboi** [comment]
https://user-images.githubusercontent.com/64006268/192623180-ef856b6e-8010-4fcf-8e58-55925cc674ad.mp4

**Pull request merged**